javascript/ql/src/semmle/javascript/Expr.qll

```diff
@@ -245,6 +245,21 @@ class Expr extends @expr, ExprOrStmt, ExprOrType, AST::ValueNode {
       ctx.(ConditionalExpr).inNullSensitiveContext()
     )
   }
+
+  /**
+   * Gets the data-flow node where exceptional data-flow will flow if this expression
+   * causes an exception to be thrown.
+   */
+  DataFlow::Node getThrowsToNode() {
```

Comment:
It makes be a little sad to see a new name introduced for this concept when it's so close to the existing concept of exceptional (return) nodes. Do we really need an API that talks about both "exceptional" nodes and "throws-to" nodes?

I'd suggest something like `getExceptionalNode` or `getExceptionalSuccessor`.

Reply:
The name I gave the method at first was actually `getExceptionalSuccessor`.  
But it felt wrong to talk about the successor of an expression.  
Data-flow nodes have successors, expressions don't. 

I'll rename it to `getExceptionalNode` for now.

Reply:
Sorry for going round in a loop here, but after some more thought the name still doesn't feel right. What do you think about `getExceptionTarget`?

Reply:
I like the sound of `getExceptionTarget`, I'll do that.